### PR TITLE
remove meaningless case in volume binding test

### PR DESCRIPTION
**Pull request by lichuqiang**
**What type of PR is this?**
/kind flake
/sig storage
/sig scheduling

**What this PR does / why we need it**:
remove meaningless case in volume binding test

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partially address #70180 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/integration/scheduler/volume_binding_test.go [hunk 1]
```diff
@@ -715,12 +715,6 @@ func TestVolumeProvision(t *testing.T) {
 			boundPvcs:       []*testPVC{{"pvc-w-canbind", classWait, ""}},
 			provisionedPvcs: []*testPVC{{"pvc-canprovision", classWait, ""}},
 		},
-		"one immediate pvc prebound, one wait provisioned": {
```
Comment (lichuqiang) [inline test/integration/scheduler/volume_binding_test.go]: I think I made this by mistake :(  pvc prebound is not guaranteed, its PV can still be bound by other pvcs.

What makes me a little curious is that in theory, the two pvcs here are created in order, they should get processed in scheduler/controller as the order too. But in fact the order got changed during events generation/consumption, the pv got bound to the pvc that is expected for provision.

Anyway, this is a meaningless case, that should be removed

**Comment by spiffxp** [comment]
/approve
I leave the /lgtm to @msau42

**Comment by spiffxp** [comment]
Note that I'm unclear whether it's just this one specific test case that is flaking in https://github.com/kubernetes/kubernetes/issues/70180... that was just the first example I ran into.  Have you looked at other failures for TestVolumeProvision to confirm this wasn't just a one-off?

**Comment by msau42** [comment]
/lgtm
We can continue to investigate if the test is failing for additional reasons

**Pull request merged**